src/command_buffer.rs

```diff
@@ -0,0 +1,225 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+use core::any::TypeId;
+use core::ptr::{self, NonNull};
+
+use crate::alloc::alloc::{alloc, dealloc, Layout};
+use crate::alloc::vec::Vec;
+use crate::archetype::TypeInfo;
+use crate::Entity;
+use crate::World;
+use crate::{align, DynamicBundle};
+
+/// Allows spawn operations to be buffered for future application to a ['World']
+///
+/// ```
+/// # use hecs::*;
+/// let mut world = World::new();
+/// let reserved_ent = world.reserve_entity();
+/// let mut entity_buffer = CommandBuffer::new();
+/// entity_buffer.spawn_at(reserved_ent,(true,0.10));
+/// entity_buffer.run_on(&mut world); // buffer can now be reused
+/// assert!(world.contains(reserved_ent));
+/// ```
+pub struct CommandBuffer {
+    entities: Vec<EntityIndex>,
+    storage: NonNull<u8>,
+    layout: Layout,
+    cursor: usize,
+    info: Vec<ComponentInfo>,
+    ids: Vec<TypeId>,
+}
+
+impl CommandBuffer {
+    /// Create an empty buffer
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    unsafe fn grow(
+        min_size: usize,
+        cursor: usize,
+        align: usize,
+        storage: NonNull<u8>,
+    ) -> (NonNull<u8>, Layout) {
+        let layout = Layout::from_size_align(min_size.next_power_of_two().max(64), align).unwrap();
+        let new_storage = NonNull::new_unchecked(alloc(layout));
+        ptr::copy_nonoverlapping(storage.as_ptr(), new_storage.as_ptr(), cursor);
+        (new_storage, layout)
+    }
+
+    unsafe fn add_inner(&mut self, ptr: *mut u8, ty: TypeInfo) {
+        let offset = align(self.cursor, ty.layout().align());
+        let end = offset + ty.layout().size();
+
+        if end > self.layout.size() || ty.layout().align() > self.layout.align() {
+            let new_align = self.layout.align().max(ty.layout().align());
+            let (new_storage, new_layout) = Self::grow(end, self.cursor, new_align, self.storage);
+            if self.layout.size() != 0 {
+                dealloc(self.storage.as_ptr(), self.layout);
+            }
+            self.storage = new_storage;
+            self.layout = new_layout;
+        }
+
+        let addr = self.storage.as_ptr().add(offset);
+        ptr::copy_nonoverlapping(ptr, addr, ty.layout().size());
+        self.info.push(ComponentInfo {
+            ty_info: ty,
+            offset,
+        });
+        self.cursor = end;
+    }
+
+    /// Buffer 'reserved_entity' with 'bundle'
```

Comment:
"reserved_entity" doesn't appear anywhere else, and the use of "buffer" as a verb here is unconventional. Additionally, you probably meant to use backticks.
```suggestion
    /// Record an entity spawn operation
```